guides/getting-started/develop-solidus.md

```diff
@@ -0,0 +1,127 @@
+# Develop Solidus
+
+If you would like to develop for Solidus and you intend to submit your work as a
+pull request, please [read the contributing guidelines][contributing] first.
+
+Getting your Solidus development environment set up is easy. First, clone the
+Solidus GitHub repo:
+
+```shell
+git clone git://github.com/solidusio/solidus.git
+```
+
+Then enter the directory you just downloaded and install Solidus's dependencies:
+
+```shell
+cd solidus
+bundle install
+```
+
+## Create a sandbox application
+
+Solidus is meant to be run within a Rails application. You can create a sandbox
+application inside the source directory that you have cloned. This gives you a
+typical Solidus store you can use for testing.
+
+By default, the sandbox includes [`solidus_auth_devise`][solidus-auth-devise],
+and the generator seeds the database and loads sample data.
+
+```shell
+bundle exec rake sandbox
+```
+
+You can prepend `DB=mysql` or `DB=postgresql` to the command in order use those
+databases instead of the default SQLite 3 database. For example:
+
+```shell
+DB=postgresql bundle exec rake sandbox
+``` 
+
+After the sandbox has been generated, you can change into its directory and
+start the server:
+
+```
+cd sandbox
+rails server
+```
+
+[contributing]: https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md
+[solidus-auth-devise]: https://github.com/solidusio/solidus_auth_devise
+
+## Testing
+
+Solidus uses [RSpec](http://rspec.info/) for testing.
+
+If you intend to submit your work to Solidus as a pull request, it must pass all
+of the Solidus test suites before it is merged. We use CircleCI to run tests on
+all incoming pull requests.
+
+To run the test suites for `solidus_frontend` and `solidus_backend`, you need to
+install [PhantomJS](http://phantomjs.org/) on your system first.
+
+You can see the build statuses [on our CircleCI status page][circleci].
+
+### Run all Solidus test suites
+
+To execute all of the test specs, run the `build.sh` script at the root of the
+Solidus project:
+
+```shell
+bash build.sh
+```
+
+This runs using PostgresQL by default, but it can be overridden by specifying
+`DB=sqlite` or `DB=mysql` in the environment. For example:
+
+```shell
+DB=mysql bash build.sh
+```
+
+Note that this will fail if you have not installed PhantomJS on your system.
+
+### Run a single test suite
+
+Each gem contains its own test suite. For example, you can run only the
+`solidus_core` gem tests within the `/core` directory:
+
+```shell
+cd core
+bundle exec rspec
+```
+
+By default, the tests run against the default SQLite 3 database. You can instead
+specify `DB=mysql` or `DB=postgresql` by prepending it to the command:
+
+```shell
+DB=postgresql bundle exec rspec
+```
+
+#### Fail fast
+
+You can enable `FAIL_FAST` to stop any test run after the first failure:
+
+```shell
+FAIL_FAST=true bundle exec rspec spec/models/state_spec.rb
```

Comment:
We should not promote a custom ENV var, but the rspec `--fail-fast` feature instead. But this is common rspec knowledge, I don't think we need to mention it in our guides at all.

Reply:
I actually took this from the [README](https://github.com/solidusio/solidus#running-an-individual-test-suite):

> You can also enable fail fast in order to stop tests at the first failure
> `FAIL_FAST=true bundle exec rspec spec/models/state_spec.rb`

Would you :+1: my removing this from the README as well?

I would agree that we should remove this if it's common rspec knowledge. Instead I'll just explicitly state to reference the rspec documentation for testing needs.